cmd/kubeadm/app/util/etcd/testing/mock_etcd_cluster.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Generated code, generated via: `mockgen -destination=./testing/mock_etcd_cluster.go -package=testing go.etcd.io/etcd/clientv3 Cluster`
+// Edited by hand for boilerplate and gofmt.
+// TODO, this should be autogenerated/autoupdated by scripts.
```

Comment:
as much as mocking the etcd cluster and ensuring behavior is a nice to have, i think we can omit this file and the new test that consumes the mock cluster for the time being.

one minor concern is that none of the kubeadm maintainers use mockgen (AFAIK) and the mocking can be done in a more familiar pattern to the existing dev team.

eventually we could have full coverage for all of the wrapped client methods, but i think we can drop this for now.

Reply:
Alright, fair enough - I usually try to opt for unit tests as a default when reasonable, but will definitely follow the standard here.

I'll re-upload a commit without the mock + unit tests.